"blinky" sample fails to build for BBC MicroBit (DT_ALIAS_LED0_GPIOS_CONTROLLER undefined)

**Describe the bug**
Following along in the Getting Started guide, I tried to build the "blinky" sample for the BBC MicroBit board. It fails due to a C compile error; apparently the constants for the LED0 pin are not defined as expected. The sample does build for the `reel_board`.

**To Reproduce**
Steps to reproduce the behavior:
1. Setup zephyr as per Getting Started Guide, steps 1-5
2. Follow step 6 by entering `west build -p auto -b bbc_microbit samples/basic/blinky`

**Expected behavior**
The build succeeds

**Impact**
To actually run the sample I need to build it for a board I have in my possession.
I would expect a minimal sample like this to build successfully for every supported board, especially since the guide doesn't specify any particular board but implies any will work.
I do own two other supported boards so I'm not dead in the water yet, but this is disappointing.

**Screenshots or console output**
```
[34/122] Building C object CMakeFiles/app.dir/src/main.c.obj
FAILED: CMakeFiles/app.dir/src/main.c.obj 
ccache /Users/snej/Couchbase/Embedded/Toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc -DBUILD_VERSION=v2.2.0-rc2-41-gb96477dd69bb -DKERNEL -DNRF51 -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/include -Izephyr/include/generated -I/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf51 -I/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/ext/hal/cmsis/Core/Include -I/Users/snej/Couchbase/Embedded/zephyrproject/modules/hal/nordic/nrfx -I/Users/snej/Couchbase/Embedded/zephyrproject/modules/hal/nordic/nrfx/drivers/include -I/Users/snej/Couchbase/Embedded/zephyrproject/modules/hal/nordic/nrfx/mdk -I/Users/snej/Couchbase/Embedded/zephyrproject/modules/hal/nordic/. -isystem /Users/snej/Couchbase/Embedded/zephyrproject/zephyr/lib/libc/minimal/include -isystem /Users/snej/Couchbase/Embedded/Toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/include -isystem /Users/snej/Couchbase/Embedded/Toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/include-fixed -Os -imacros/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mthumb -mcpu=cortex-m0 -imacros/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/samples/basic/blinky=CMAKE_SOURCE_DIR -fmacro-prefix-map=/Users/snej/Couchbase/Embedded/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/Users/snej/Couchbase/Embedded/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -mabi=aapcs -march=armv6s-m -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj   -c /Users/snej/Couchbase/Embedded/zephyrproject/zephyr/samples/basic/blinky/src/main.c
/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/samples/basic/blinky/src/main.c: In function 'main':
/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/samples/basic/blinky/src/main.c:24:27: error: 'DT_ALIAS_LED0_GPIOS_CONTROLLER' undeclared (first use in this function); did you mean 'DT_ALIAS_SW0_GPIOS_CONTROLLER'?
   24 |  dev = device_get_binding(DT_ALIAS_LED0_GPIOS_CONTROLLER);
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                           DT_ALIAS_SW0_GPIOS_CONTROLLER
/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/samples/basic/blinky/src/main.c:24:27: note: each undeclared identifier is reported only once for each function it appears in
/Users/snej/Couchbase/Embedded/zephyrproject/zephyr/samples/basic/blinky/src/main.c:29:32: error: 'DT_ALIAS_LED0_GPIOS_PIN' undeclared (first use in this function); did you mean 'DT_ALIAS_SW0_GPIOS_PIN'?
   29 |  ret = gpio_pin_configure(dev, DT_ALIAS_LED0_GPIOS_PIN,
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
      |                                DT_ALIAS_SW0_GPIOS_PIN
[47/122] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/clock_control/nrf_power_clock.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /Users/snej/Couchbase/Embedded/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: macOS 10.15.3
 - Toolchain: GCC-ARM
 - Commit: b96477dd69bbc6b1830048e8eee59cf62c3684da (HEAD -> master, origin/master)

--- a/doc/getting_started/index.rst
+++ b/doc/getting_started/index.rst
@@ -361,6 +361,12 @@ The sample :ref:`blinky-sample` blinks an LED on the target board.  By
 building and running it, we can verify that the environment and tools
 are properly set up for Zephyr development.
 
+   .. note:: This sample is compatible with most boards supported by
+      Zephyr, but not all of them. See the :ref:`blinky sample requirements
+      <blinky-sample-requirements>` for more information. If this sample is not
+      compatible with your board, a good alternative to try is the
+      :ref:`Hello World sample <hello_world>`.
+
 #. Set build environment variables:
 
    .. tabs::

--- a/doc/guides/dts/intro.rst
+++ b/doc/guides/dts/intro.rst
@@ -271,6 +271,8 @@ reg
     ``reg = <0x40003000 0x1000>;`` means the register map occupies 0x1000 bytes
     in the memory map.
 
+.. _devicetree-in-out-files:
+
 Input and output files
 **********************
 

--- a/samples/basic/blinky/README.rst
+++ b/samples/basic/blinky/README.rst
@@ -1,7 +1,7 @@
 .. _blinky-sample:
 
-Blinky Application
-##################
+Blinky sample
+#############
 
 Overview
 ********
@@ -10,22 +10,24 @@ The Blinky example shows how to configure GPIO pins as outputs which can also be
 used to drive LEDs on the hardware usually delivered as "User LEDs" on many of
 the supported boards in Zephyr.
 
+.. _blinky-sample-requirements:
+
 Requirements
 ************
 
 The demo assumes that an LED is connected to one of GPIO lines. The
-sample code is configured to work on boards that have defined the led0
-alias in their board devicetree description file. Doing so will generate
-these variables:
+sample code is configured to work on boards that have defined the ``led0``
+alias in their :ref:`board's devicetree description file
+<devicetree-in-out-files>`, :file:`<board>.dts`.
+Doing so will generate these variables:
 
-- DT_ALIAS_LED0_GPIOS_CONTROLLER
-- DT_ALIAS_LED0_GPIOS_PIN
-
+- ``DT_ALIAS_LED0_GPIOS_CONTROLLER``
+- ``DT_ALIAS_LED0_GPIOS_PIN``
 
 Building and Running
 ********************
 
-This samples does not output anything to the console.  It can be built and
+This sample does not output anything to the console. It can be built and
 flashed to a board as follows:
 
 .. zephyr-app-commands::

--- a/samples/basic/blinky/src/main.c
+++ b/samples/basic/blinky/src/main.c
@@ -12,7 +12,19 @@
 #define SLEEP_TIME_MS   1000
 
 #ifndef DT_ALIAS_LED0_GPIOS_FLAGS
-#define DT_ALIAS_LED0_GPIOS_FLAGS 0
+#define FLAGS 0
+#else
+#define FLAGS DT_ALIAS_LED0_GPIOS_FLAGS
+#endif
+
+/* Make sure the board's devicetree declares led0 in its /aliases. */
+#ifdef DT_ALIAS_LED0_GPIOS_CONTROLLER
+#define LED0	DT_ALIAS_LED0_GPIOS_CONTROLLER
+#define PIN	DT_ALIAS_LED0_GPIOS_PIN
+#else
+#error "Unsupported board: led0 devicetree alias is not defined"
+#define LED0	""
+#define PIN	0
 #endif
 
 void main(void)
@@ -21,20 +33,18 @@ void main(void)
 	bool led_is_on = true;
 	int ret;
 
-	dev = device_get_binding(DT_ALIAS_LED0_GPIOS_CONTROLLER);
+	dev = device_get_binding(LED0);
 	if (dev == NULL) {
 		return;
 	}
 
-	ret = gpio_pin_configure(dev, DT_ALIAS_LED0_GPIOS_PIN,
-				 GPIO_OUTPUT_ACTIVE
-				 | DT_ALIAS_LED0_GPIOS_FLAGS);
+	ret = gpio_pin_configure(dev, PIN, GPIO_OUTPUT_ACTIVE | FLAGS);
 	if (ret < 0) {
 		return;
 	}
 
 	while (1) {
-		gpio_pin_set(dev, DT_ALIAS_LED0_GPIOS_PIN, (int)led_is_on);
+		gpio_pin_set(dev, PIN, (int)led_is_on);
 		led_is_on = !led_is_on;
 		k_sleep(SLEEP_TIME_MS);
 	}

